lib/mvi/service.rb

```diff
@@ -32,6 +32,19 @@ class Service < Common::Client::Base
 
     STATSD_KEY_PREFIX = 'api.mvi' unless const_defined?(:STATSD_KEY_PREFIX)
 
+    def find_profile_with_attributes(user_attributes)
+      raw_response = with_monitoring do
+        perform(
+          :post,
+          '',
+          message_user_attributes(OpenStruct.new(user_attributes)),
+          soapaction: OPERATIONS[:find_profile]
+        )
+      end
+
+      MVI::Responses::FindProfileResponse.with_parsed_response(raw_response)
+    end
```

Comment:
hmmm, i'm wondering if it might be a good idea to consider refactoring MVI to better support this. @kreek do you have any thoughts on this approach?

Reply:
We're essentially reimplementing `find_profile` (as `find_profile_with_attributes`) and then moving its error handling up to the HCA model. It would be nice if we could reuse `find_profile` instead by redefining `create_profile_message`. The tricky part is that `find_profile` uses the `Rails.logger.measure_info` block which uses `logging_context` that expects a user. One way could be to define `measure_info` in `MVI::Service`:
```ruby
# lib/mvi/service.rb
def find_profile(user)
  with_monitoring do
    measure_info(user) do
      raw_response = perform(:post, '', create_profile_message(user), soapaction: OPERATIONS[:find_profile])
      MVI::Responses::FindProfileResponse.with_parsed_response(raw_response)
    end
  end
rescue Breakers::OutageException => e
# ...

def measure_info(user)
  Rails.logger.measure_info('Performed MVI Query', payload: logging_context(user)) { yield }
end
```
and then redefine `create_profile_message` and `measure_info` in the subclass:
```ruby
# lib/mvi/attr_service.rb
module MVI
  class AttrService < Service
    configuration MVI::Configuration

    private

    def create_profile_message(user_attributes)
      message_user_attributes(OpenStruct.new(user_attributes))
    end

    def measure_info(_user_attributes)
      Rails.logger.measure_info('Performed MVI Query') { yield }
    end
  end
end
```
then you can call `find_profile` with user attributes from HCA
```ruby
MVI::AttrService.new.find_profile(user_attributes(form)).profile.icn
```
Does add a class but allows the existing error handling to be used.

Reply:
thanks for the suggestion, i changed it